service/matching/fairness_util.go

```diff
@@ -24,3 +24,29 @@ func getEffectiveWeight(overrides fairnessWeightOverrides, pri *commonpb.Priorit
 	}
 	return weight
 }
+
+func mergeFairnessWeightOverrides(
+	existing map[string]float32,
+	set map[string]float32,
+	unset []string,
+	maxFairnessKeyWeightOverrides int,
+) (map[string]float32, error) {
+	res := make(map[string]float32, len(existing))
+	for k, v := range existing {
+		res[k] = v
+	}
```

Comment:
```suggestion
	res := maps.Clone(existing)
```

Reply:
This caused a panic actually since before it handled `existing` being `nil` but `Clone` returns `nil` instead. Saw it in the func tests now.

Reply:
Good to know we have that - I've changed it to return the `set` now in that case. Added a test as well.